test/integration/targets/ansible-test-sanity/ansible_collections/ns/col/meta/runtime.yml

```diff
@@ -2,4 +2,11 @@ requires_ansible: '>=2.11'  # force ansible-doc to check the Ansible version (re
 plugin_routing:
   modules:
     hi:
-      redirect: hello
+      redirect: ns.col2.hello
+    hiya:
+      redirect: ns.col2.package.subdir.hiya
+  module_utils:
+    hi:
+      redirect: ns.col2
```

Comment:
Looks good to me, just wondered about the module_utils change.

It looks like this would be an error https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/module_common.py#L714-L715

But if a redirect starts with `ansible_collections`, a valid collection name or fqcr isn't required. Maybe these should stay str type for now? Or just enforce fqcr if it doesn't start with `ansible_collections`.

Reply:
The thing is that https://github.com/ansible/ansible/blob/devel/lib/ansible/config/ansible_builtin_runtime.yml#L8796-L8797 contains exactly such an entry (`dellemc.openmanage`).

But yeah, keeping it `str` for now is probably best.

Reply:
It looks the redirect initially was 
```
    remote_management.dellemc:
      redirect: community.general.remote_management.dellemc
```
and was changed in ca448f7c350fae8080dcfec648342d2fc8837da0  since the dellemc.openmanage collection doesn't have a top level dir for its module_utils anymore. But I don't think it ever worked.

We force `module_utils/__init__.py` to be empty in a collection https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/module_common.py#L832-L833 so I'm not sure if redirecting to the package makes sense. FWIW, from a brief search I do not see anything that would have used that redirect pre or post collection split.

Reply:
> But I don't think it ever worked.

That's very likely :)

> We force `module_utils/__init__.py` to be empty in a collection https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/module_common.py#L832-L833 so I'm not sure if redirecting to the package makes sense.

Since that sanity test can be ignored (and there are valid use-cases where it is ignored, at least for sub-packages of `plugins.module_utils`), I wouldn't say "force", but "encourage" :)

Reply:
Subpackages of plugins.module_utils can contain stuff in their `__init__`, but the top level package appears to be forced empty, at least from my testing. TIL there is a sanity test encouraging the same for subpackages :)

Reply:
It's this one: https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_util/controller/sanity/code-smell/empty-init.py

I wasn't aware that `plugins.module_utils` itself is special, though it's not very surprising. I've never had a reason to actually try that though :)